### PR TITLE
Add typed calls for concrete Swift functions

### LINKED ISSUE #28: Invoke concrete Swift functions with Swift calling conventions
## Problem or goal

Concrete Swift functions use Swift calling conventions, ownership, and value lowering that cannot be inferred from a C ABI call interface alone.

## Expected outcome

Provide `runtime.swiftFunction(named:as:in:)` for concrete synchronous, nonthrowing Swift declarations. Resolve source-level names through shared image indexes, derive or explicitly describe their Swift ABI, and return reusable typed invocation handles.

## Acceptance criteria

- Use function-type metatypes and ordinary values without requiring mangled names or raw address casts.
- Account for Swift argument/result lowering, direct and indirect results, ownership, and architecture-specific registers rather than treating arbitrary Swift signatures as C signatures.
- Support known concrete scalar/object/value representations and provide a clear adapter boundary for foreign layouts.
- Verify real Swift fixture functions with zero and many mixed arguments, optional objects, value results, image lifetime, and unsupported signatures on each supported calling-convention implementation.
- Keep generic metadata construction, opaque results, async, throwing functions, and complex existential containers outside the initial contract.
- Document exact runtime evidence and build-only platform coverage in the PR; do not infer device execution guarantees from compilation.

Part of #1. This separates Swift ABI implementation from the typed C/C++ frontend in #4 and the native adapters in #6.

### PR DESCRIPTION
**Pull request by lynnswap**
## Purpose

Call concrete synchronous, nonthrowing Swift functions by source-level name with ordinary Swift types and values. Closes #28; part of #1.

## Changes

- Add `swiftFunction(named:as:in:)` and reusable typed invocation handles, including label-only names whose canonical Swift types come from the supplied metatype.
- Lower arguments and results with Swift calling conventions on arm64, arm64_32, and x86_64, including mixed register/stack arguments and direct/indirect results.
- Preserve guaranteed object/String arguments and owned results, support explicit trivial value adapters, and retain implementation images.
- Reject declared generic, async, throwing, inout, and consuming conventions; document the adapter boundary for other representations.
- Add quick-start and DocC guidance plus an external Swift-library consumer and guarded-memory boundary tests.

## Testing

- Xcode 26.6 / Swift 6.3.3: the package suite passed on macOS arm64.
- All 13 Swift invocation/boundary cases passed on macOS arm64 and x86_64, and in Release on arm64, including mixed integer/Float components in the same storage chunk.
- All seven external consumers passed, including calls into an unimported Swift library after releasing its original loader reference.
- iOS, visionOS, watchOS, and tvOS device-target builds passed. Arm64_32 compiler output verifies direct three-register Int64 aggregates; arm64e backend compilation and relocatable linking passed. Device runtime behavior remains unverified.
- DocC build, shellcheck, and complete diff whitespace checks passed.
- Local Codex review: no findings.

### CODE REVIEW

File: Sources/ABIBridge/SwiftValues.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        if isAdapter {
+            let codec = try CValueCodec<Value>()
+            cValue = codec
+            type = codec.type
+        } else if isObject {
+            type = try CValueType(scalar: ABIValuePointer)
+            cValue = nil
+        } else if Value.self == String.self {
+            let word = try CValueType(scalar: MemoryLayout<UInt>.size == 8 ? ABIValueUInt64 : ABIValueUInt32)
+            type = try CValueType(fields: Array(repeating: word, count: MemoryLayout<String>.size / MemoryLayout<UInt>.size))
```
[the remark this answers is not in the document]
Comment (lynnswap) [reply Sources/ABIBridge/SwiftValues.swift:29]: Keeping the current representation: Xcode 26.6 / Swift 6.3.3 emits three i32 components for String on arm64_32. The checked-in `Tests/NativeConsumer/SwiftFixture.swift` produces this signature for `decorate(_:)`:

```llvm
define swiftcc { i32, i32, i32 } @"$s20SwiftFunctionFixture8decorateyS2SF"(i32 %0, i32 %1, i32 %2)
```

Reproduce with the selected Xcode 26.6 toolchain:

```sh
xcrun swiftc -parse-as-library -module-name SwiftFunctionFixture -enable-library-evolution -O -target arm64_32-apple-watchos11.4 -sdk "$(xcrun --sdk watchos --show-sdk-path)" -emit-ir Tests/NativeConsumer/SwiftFixture.swift -o /tmp/swift-fixture.ll
```

The UInt64-pair `rawBits` helper converts the 32-bit stored representation into a different format; see the [32-bit implementation](https://github.com/swiftlang/swift/blob/ee94684595839400b25107483954d39e6112b8af/stdlib/public/core/StringObject.swift#L259). The proposed UInt64 components would disagree with the emitted call ABI.

File: Sources/ABIBridgeCore/SwiftInvocation.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    flatten(type, 0, fields);
+    Layout result;
+    // Swift coalesces adjacent integer storage inside a pointer-sized chunk;
+    // floating fields stay separate. See Clang's SwiftAggLowering and the Swift
+    // ABI CallingConventionSummary, rather than the platform C aggregate rules.
+    for (const auto &field : fields) {
+        if (!result.components.empty()) {
+            auto &previous = result.components.back();
+            if (!previous.floating && !field.floating &&
+                (previous.offset + previous.size - 1) / sizeof(void *) == field.offset / sizeof(void *)) {
```
[the remark this answers is not in the document]
Comment (lynnswap) [reply Sources/ABIBridgeCore/SwiftInvocation.cpp:70]: Verified that Swift emits `swiftcc { i8, float } (i8, float)` for this layout on both arm64 and x86_64. Added a [runtime comparison against a normal Swift call](https://github.com/lynnswap/ABIBridge/blob/d3c08ecef0dc4e2022607b236d1c1bc05a341b51/Tests/ABIBridgeTests/SwiftCallBoundaryTests.swift) in d3c08ec; it passes in Debug and Release on arm64 and on x86_64. The existing separate integer/floating banks match that ABI, so the lowering remains unchanged.

**Review by lynnswap — COMMENTED** [review]
@codex review
